src/components/TimePicker/pageObject/index.js

```diff
@@ -0,0 +1,237 @@
+            .isFocused();
+    }
+
+    /**
+     * Returns true when the hour input has focus.
+     * @method
+     * @returns {bool}
+     */
+    hasFocusHourInput() {
+        return $(this.timePickerModal)
+            .$('input[data-id="hour"]')
+            .isFocused();
+    }
+
+    /**
+     * Get the hour input value.
+     * @method
+     * @returns {string}
+     */
+    getHourValue() {
+        return $(this.timePickerModal)
+            .$('input[data-id="hour"]')
+            .getValue();
+    }
+
+    /**
+     * Returns true when the minutes input has focus.
+     * @method
+     * @returns {bool}
+     */
+    hasFocusMinutesInput() {
+        return $(this.timePickerModal)
+            .$('input[data-id="minutes"]')
+            .isFocused();
+    }
+
+    /**
+     * Get the minutes input value.
+     * @method
+     * @returns {string}
+     */
+    getMinutesValue() {
+        return $(this.timePickerModal)
+            .$('input[data-id="minutes"]')
+            .getValue();
+    }
+
+    /**
+     * Returns true when the am-pm selector has focus.
+     * @method
+     * @returns {bool}
+     */
+    hasFocusAmPmSelect() {
+        return $(this.timePickerModal)
+            .$('fieldset[data-id="fieldset-element"]')
+            .isFocused();
+    }
+
+    /**
+     * Returns true when the am input is selected.
+     * @method
+     * @returns {bool}
+     */
+    isAmInputSelected() {
+        browser.waitUntil(() =>
+            $(this.timePickerModal)
+                .$('fieldset[data-id="fieldset-element"]')
+                .isFocused(),
+        );
+        return $(this.timePickerModal)
+            .$$('input[type="radio"]')[0]
+            .isSelected();
+    }
+
+    /**
+     * Returns true when the pm input is selected.
+     * @method
+     * @returns {bool}
+     */
+    isPmInputSelected() {
+        browser.waitUntil(() =>
+            $(this.timePickerModal)
+                .$('fieldset[data-id="fieldset-element"]')
+                .isFocused(),
+        );
+        return $(this.timePickerModal)
+            .$$('input[type="radio"]')[1]
+            .isSelected();
+    }
+
+    /**
+     * Type in the hour input element.
+     * @method
+     * @param {string} value - The value to type in the hour input element.
+     */
+    setHourValue(value) {
+        $(this.timePickerModal)
+            .$('input[data-id="hour"]')
+            .setValue(value);
+    }
+
+    /**
+     * Type in the minutes input element.
+     * @method
+     * @param {string} value - The value to type in the minutes input element.
+     */
+    setMinutesValue(value) {
+        $(this.timePickerModal)
+            .$('input[data-id="minutes"]')
+            .setValue(value);
+    }
+
+    /**
+     * Wait until the TimePicker is open.
+     * @method
+     */
+    waitUntilOpen() {
+        new PageModal(this.timePickerModal).waitUntilOpen();
+        browser.waitUntil(() => this.isTimePickerOpen());
```

Comment:
Since inside modal we made browser.waitUntil why repeat here?
I think this can be simplified to:
```
waitUntilOpen() {
    this.modalPage. waitUntilOpen();
}
```